pkg/mountstatus/mountstatus.go

```diff
@@ -0,0 +1,65 @@
+package mountstatus
+
+import (
+	"sync"
+
+	"github.com/containerd/containerd/reference/docker"
+)
+
+// ImagePullStatus represents mount status of an image
+type ImageMountStatus int
+
+// https://stackoverflow.com/questions/14426366/what-is-an-idiomatic-way-of-representing-enums-in-go
+const (
+	// StatusNotFound means there has been no attempt to mount the image
+	StatusNotFound ImageMountStatus = -1
+	// StillMounting means the image is still being mounted as a volume
+	StillMounting ImageMountStatus = iota
+	// Mounted means the image has been mounted as a volume
+	Mounted
+	// Errored means there was an error during image mount
+	Errored
+)
+
+// ImageMountStatusRecorder records the status of image mounts
+type ImageMountStatusRecorder struct {
+	status map[docker.Named]ImageMountStatus
+	mutex  sync.Mutex
```

Comment:
it seems like all you want is thread-safe map ? may-be you can use this directly - https://pkg.go.dev/sync#Map

Reply:
Interesting thought but I don't think it's a good idea because we don't get type safety. 

> The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.

> (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or

This is not completely true in our case because we are making multiple writes for the same key.